src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -1885,6 +1931,34 @@ public void headersAreSetAndSentThroughBuilder() {
         assertEquals("ExampleValue", header.get());
     }
 
+    @Test
+    public void headersAreNotClearedWhenConnectionIsRestarted() {
```

Comment:
I think there should be another test that is a combination of the two just added. Where user sets an Authorization header explicitly and then there is a negotiate redirect with a new header value, and then stopping and starting again preserves the original user header